### ISSUE TITLE
Unable to apply "width parameter" to EnumerationBuilder.create() method in python API

### ISSUE THREAD

**Issue by pr0xy-t** [Vector35/binaryninja-api#3424]
**Version and Platform (required):**
- Binary Ninja Version: 3.1.3644-dev
- OS: Ubuntu Linux
- OS Version: 22.04
- CPU Architecture x64

**Bug Description:**
This bug is caused by Python API implementation. (python/binaryninja/types.py#EnumerationBuilder.create)
I (want to) create an enum type instance with width=4 using Python API's EnumerationBuilder.create(width=4) method.
But EnumerationBuilder.create method return width=8 instance.

**Steps To Reproduce:**
```
>>> enum_type_fd = EnumerationBuilder.create(members=[], width=4, arch=Architecture["x86_64"], sign=True, platform=Platform["linux-x86_64"])
>>> enum_type_fd.width
8
```

**Expected Behavior:**
```
>>> enum_type_fd = EnumerationBuilder.create(members=[], width=4, arch=Architecture["x86_64"], sign=True, platform=Platform["linux-x86_64"])
>>> enum_type_fd.width
4
```

**Screenshots:**
![Screenshot from 2022-08-22 20-19-00](https://user-images.githubusercontent.com/42575435/185912106-3c9404e2-4fd1-44de-b511-ed9ce153c7fd.png)

**Additional Information:**
This bug is caused by overwritten width parameter in create method.
The width parameter being passed to _width in the EnumerationBuilder.create method (Line 349) and next then overwritten by arch.address_size (Line 351) or the constant 4 (Line 353).
![Screenshot from 2022-08-22 20-22-50](https://user-images.githubusercontent.com/42575435/185912066-24f13a67-2f74-4939-8f2a-afbce06a555a.png)

**Comment by plafosse** [comment]
Thanks for the bug report. This should be fixed in 3.1.3645. We actually have extensive unit tests on this stuff but somehow missed that edge case!

**Comment by pr0xy-t** [comment]
Thank you for your quick response!!!
